app/boards/arm/corneish_zen/Kconfig.board

```diff
@@ -3,10 +3,18 @@
 # SPDX-License-Identifier: MIT
 #
 
+config BOARD_CORNEISH_ZEN_V1_LEFT
```

Comment:
To avoid all the annoying `||` in all the left/right checks, it might be worthwhile to create a `BOARD_CORNEISH_ZEN_LEFT` that gets `select`ed by the V1 and V2 actual boards, then use those abstract Kconfig settings in the checks everywhere.

Reply:
I tried to follow the [Kyria approach](https://github.com/zmkfirmware/zmk/blob/main/app/boards/shields/kyria/Kconfig.defconfig) but it resulted in circular dependencies; I think `Kconfig.board` is processed differently compared to `Kconfig.shield` files. I'll try to find a board example if I can and see if I can refactor some other way 👍

Reply:
I added left and right Kconfigs following [`efr32_radio` board in Zephyr](https://github.com/zmkfirmware/zephyr/blob/v3.0.0%2Bzmk-fixes/boards/arm/efr32_radio/Kconfig.board), that cleaned it up a little.